### PR TITLE
Rework of FederatedSchemaPrinter fix

### PR DESCRIPTION
**Pull request by sungam3r**
See #2576. I think that it should be done in different way. First - type for queries may have any name, not only "Query".  Second - empty type definitions are allowed by the spec. See https://github.com/graphql-dotnet/parser/pull/117 fix. I changed both printers and test.

### CODE REVIEW

**Review by Shane32 — APPROVED** [review]
I don't use federation.  If this seems to be better suited to the spec, it's fine with me.

File: src/GraphQL/Utilities/Federation/FederatedSchemaPrinter.cs [hunk 1]
```diff
@@ -68,7 +64,10 @@ public override string PrintObject(IObjectGraphType type)
 
             var extended = isExtension ? "extend " : "";
 
-            return FormatDescription(type.Description) + "{1}type {2}{3}{4} {{{0}{5}{0}}}".ToFormat(Environment.NewLine, extended, type.Name, implementedInterfaces, federatedDirectives, PrintFields(type));
+            if (type.Fields.Any(x => !IsFederatedType(x.ResolvedType!.GetNamedType().Name)))
```
Comment (sungam3r) [inline src/GraphQL/Utilities/Federation/FederatedSchemaPrinter.cs:67]: `FederatedSchemaPrinter` overrides `PrintFields` so this check correlates to `PrintFields` implementation.

**Comment by sungam3r** [comment]
@myermian Could you please check this fix? If OK I would like to merge PR as "real" SDL fix.

**Comment by sungam3r** [comment]
So... I'll merge it. I hope that fix is correct.

**Pull request merged**